oracle/src/feeds/kraken.rs

```diff
@@ -0,0 +1,84 @@
+use super::{get_http, PriceFeed};
+use crate::{config::CurrencyStore, currency::*, Error};
+use async_trait::async_trait;
+use clap::Parser;
+use reqwest::Url;
+use serde_json::Value;
+
+#[derive(Parser, Debug, Clone)]
+pub struct KrakenCli {
+    /// Fetch the exchange rate from Kraken
+    #[clap(long)]
+    kraken_url: Option<Url>,
+}
+
+pub struct KrakenApi {
+    url: Url,
+}
+
+impl Default for KrakenApi {
+    fn default() -> Self {
+        Self {
+            url: Url::parse("https://api.kraken.com/0").unwrap(),
+        }
+    }
+}
+
+fn extract_response<'a>(value: &'a Value) -> Option<&'a str> {
```

Comment:
Can you add a unit test with for this with a hardcoded response? Ideally we'd have that for all feeds